pkg/didcomm/protocol/didexchange/states.go

```diff
@@ -136,17 +142,25 @@ func (s *requested) CanTransitionTo(next state) bool {
 	return stateNameResponded == next.Name()
 }
 
-func (s *requested) Execute(msg dispatcher.DIDCommMsg) (state, error) {
+func (s *requested) Execute(msg dispatcher.DIDCommMsg, outbound dispatcher.Outbound, doc *did.Doc) (state, error) {
 	switch msg.Type {
 	case ConnectionInvite:
 		if msg.Outbound {
 			return nil, fmt.Errorf("outbound invitations are not allowed for state %s", s.Name())
 		}
 		// send did-exchange Request
+		err := sendExchangeRequest(msg, outbound, doc)
```

Comment:
We're missing something here - I don't know if you want to leave a TODO or do it in this PR. But it needs to be fixed:

`Service.Handle()` is using the `ID` from the Invitation as the threadID, when in this particular scenario (outbound Request) it should be the request's ID.

Reply:
@llorllale are you referring to the request that is created from an inbound invitation ?